homeassistant/components/scrape/config_flow.py

```diff
@@ -47,58 +58,110 @@
         TextSelectorConfig(type=TextSelectorType.PASSWORD)
     ),
-    vol.Optional(CONF_HEADERS): ObjectSelector(),
-    vol.Optional(CONF_UNIT_OF_MEASUREMENT): TextSelector(),
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): BooleanSelector(),
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): NumberSelector(
+        NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+    ),
+    vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): NumberSelector(
+        NumberSelectorConfig(min=15, step=15, mode=NumberSelectorMode.BOX)
+    ),
+}
+
+SENSOR_SETUP_OPT = {
+    vol.Optional(CONF_ATTRIBUTE): TextSelector(),
+    vol.Optional(CONF_INDEX, default=0): NumberSelector(
+        NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+    ),
+    vol.Required(CONF_SELECT): TextSelector(),
+    vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
     vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[e.value for e in SensorDeviceClass],
+            options=[cls.value for cls in SensorDeviceClass],
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
     vol.Optional(CONF_STATE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[e.value for e in SensorStateClass],
+            options=[cls.value for cls in SensorStateClass],
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    ),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
+        SelectSelectorConfig(
+            options=[TEMP_CELSIUS, TEMP_FAHRENHEIT],
+            custom_value=True,
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
-    vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
-    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): BooleanSelector(),
 }
 
-DATA_SCHEMA = vol.Schema({**SCHEMA_SETUP, **SCHEMA_OPT})
-DATA_SCHEMA_OPT = vol.Schema({**SCHEMA_OPT})
+SENSOR_SETUP = {
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
+}
+
+
+def return_sensor_step(user_input: dict[str, Any]) -> str:
+    """Return sensor step."""
+    return "sensor"
+
+
+def validate_rest_setup(user_input: dict[str, Any]) -> dict[str, Any]:
+    """Validate rest setup."""
+    hass = async_get_hass()
+    rest_config = {
+        **dict(user_input),
+        CONF_SCAN_INTERVAL: int(user_input[CONF_SCAN_INTERVAL]),
+        CONF_TIMEOUT: int(user_input[CONF_TIMEOUT]),
+    }
+    try:
+        rest = create_rest_data_from_config(hass, rest_config)
+        asyncio.run_coroutine_threadsafe(rest.async_update(), hass.loop)
+    except Exception as err:
+        raise SchemaFlowError("resource_error") from err
+    return user_input
+
+
+DATA_SCHEMA_RESOURCE = vol.Schema(RESOURCE_SETUP)
+DATA_SCHEMA_SENSOR = vol.Schema({**SENSOR_SETUP, **SENSOR_SETUP_OPT})
+DATA_SCHEMA_SENSOR_OPT = vol.Schema(SENSOR_SETUP_OPT)
 
 CONFIG_FLOW: dict[str, SchemaFlowFormStep | SchemaFlowMenuStep] = {
```

Comment:
For `async_config_entry_title`, you will need to adjust to return `options[CONF_RESOURCE]`.
You could also use `options["sensor"][0][CONF_NAME]` but I think using the URL is easier.

For `async_config_flow_finished`, you should probably `pop` the sensor config:
```python
        data: dict[str, Any] = dict(options)
        data.pop("sensor")
        self._async_abort_entries_match(data)
```

Reply:
If I pop the sensor then it would never match with the config entry so I would have to leave it for testing a full match.

Reply:
I don't understand your comment.
If you pop `sensors`, then it should match on (resource + method + auth + user + pass + headers + ssl + timeout)
The matching is done on all items passed in `match_dict`, even if there are extra items in the options.

But ultimately maybe it should abort after "user" step not after "sensor" step.

Maybe we should be even more explicit about what goes into the matching dict
makes it match". SSL and timeout shouldn't be part of matching dict.
```python
        matching_keys: set[str] = {
            CONF_RESOURCE,
            CONF_METHOD,
            CONF_AUTHENTICATION,
            CONF_USERNAME,
            CONF_PASSWORD,
            CONF_HEADERS,
        }
        data: dict[str, Any] = {k: v for k, v in options.items() if k in matching_keys}
        self._async_abort_entries_match(data)
```

Reply:
I was reading the test in the function backwards so it's clear.
Good suggestion also to test only against specific keys